modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/resource/test/HeadlessBuilderResourceTest.java

```diff
@@ -1717,6 +1845,73 @@ public void testPostObjectEntry() throws Exception {
 		Assert.assertEquals(objectEntries.toString(), 1, objectEntries.size());
 	}
 
+	@Test
+	public void testPostWithObjectException() throws Exception {
+		ObjectDefinition objectDefinition = _addObjectDefinition(
+			8, true, ObjectDefinitionConstants.SCOPE_COMPANY);
+
+		_addAPIApplicationWithPostEndpoint(
+			true, _API_APPLICATION_ERC_1, _API_ENDPOINT_ERC_1, _BASE_URL_1, 8,
+			objectDefinition.getExternalReferenceCode(), "/test",
+			APIApplication.Endpoint.Scope.COMPANY);
+
+		_publishAPIApplication(_API_APPLICATION_ERC_1);
+
+		JSONAssert.assertEquals(
+			JSONUtil.put(
+				"status", "BAD_REQUEST"
+			).put(
+				"title",
+				"No value was provided for required object field " +
+					"\"richTextField\""
+			).toString(),
+			HTTPTestUtil.invokeToJSONObject(
+				JSONUtil.put(
+					"booleanProperty", RandomTestUtil.randomBoolean()
+				).toString(),
+				StringBundler.concat("c/", _BASE_URL_1, "/test"),
+				Http.Method.POST
+			).toString(),
+			JSONCompareMode.STRICT);
+	}
+
+	@Test
+	public void testPostWithoutResponseSchema() throws Exception {
```

Comment:
So I could merge it but it will make the other test structure more complex because I would need to modify the creation of the API Application so that it also includes an endpoint without response schema. I think it's better leaving this as a separate test.

Reply:
hey guys I split the tests again to keep consistency with the get ones. Let's see what Brian says.